src/docs/release/breaking-changes/rendereditable-layout-before-hit-test.md

```diff
@@ -0,0 +1,84 @@
+---
+title: The RenderEditable needs to be laid out before hit testing
+description: The hit testing of RenderEditable requires additional information that is only available after the layout.
+---
+
+## Summary
+
+The `RenderEditable` needs to be laid out in order to process hit
+testing; Otherwise, the framework throws an assertion error indicates
+the `RenderEditable` has not been laid out.
+`Failed assertion: line xxxx pos xx: '!debugNeedsLayout': is not true.`
+
+## Context
+
+To support gesture recognizers in selectable text, the
+`RenderEditable` requires the layout information about the
+text spans to determine which text span receive the
+pointer event. Therefore, we made the layout to be a hard
+requirement for performing the hit testing to the
+`RenderEditable`.
+
+## Description of change
+
+The `RenderEditable` needs to be laid out before hit testing.
+
+## Migration guide
+
+If you are seeing `'!debugNeedsLayout': is not true.` assertion error while hit testing the
+`RenderEditable`, make sure to layout the `RenderEditable` before doing so.
+
+Code before migration:
+
+<!-- skip -->
+```dart
+test('attach and detach correctly handle gesture', () {
+  final RenderEditable editable = RenderEditable(
+    //...
+  );
+  final PipelineOwner owner = PipelineOwner(onNeedVisualUpdate: () { });
+  editable.attach(owner);
+  // This should throw an assertion error because the render editable has not been laid out.
+  editable.handleEvent(const PointerDownEvent(), BoxHitTestEntry(editable, const Offset(10,10)));
+  editable.detach();
+});
+```
+
+Code after migration:
+
+<!-- skip -->
```

Comment:
see if you can remove the skip somehow. it'd be good to syntax check the after code

Reply:
@Hixie, it occurs to me that removing the skip tag on new API will fail, because Travis checks the code against the stable release, or 1.12. Also, the latest commit DID cause Travis to fail, presumably from the new code.